packages/core/src/components/toast/toaster.tsx

```diff
@@ -105,14 +117,22 @@ export class Toaster extends AbstractPureComponent<IToasterProps, IToasterState>
     // auto-incrementing identifier for un-keyed toasts
     private toastId = 0;
 
-    public show(props: IToastProps) {
-        const options = this.createToastOptions(props);
+    public show(props: IToastProps, key?: string) {
+        const options = this.createToastOptions(props, key);
         this.setState(prevState => ({
             toasts: [options, ...prevState.toasts],
         }));
         return options.key;
     }
 
+    public showOrUpdate(key: string, props: IToastProps) {
```

Comment:
i wonder if this method (and, by extension, `update`) is really necessary or it it can all be accomplished through `show(props, key)`?

Reply:
Agree that this could replace `show()`. The semantics for `update()` are different - `update()` may not result in a toast being show, while `show()` and `showOrUpdate()` always do. That said, I'm not sure the semantics of `update()` are all that useful.

Reply:
I kinda dig the single `show` method that'd do an update if the key already exists, seems pretty natural no? Also with the code below, I feel like `showOrUpdate` should actually be called `updateOrShow` for clarity